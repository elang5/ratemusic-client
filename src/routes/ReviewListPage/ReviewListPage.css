.album-reviews-page {
  display: flex;
  flex-flow: column wrap;
  align-items: center;
}

.review-list {
  padding: 0;
  list-style-type: none;
  width: 70%;
}

.post-review {
  text-decoration: none;
  color: white;
  font-size: 20px;
  margin: 2rem 0;
}

.review-btn, .listen-btn {
  border: none;
  padding: 12px 14px;
  font-size: 20px;
  border-radius: 5px;
  cursor: pointer;
}

.listen-btn {
  background: #1DB954;
  color: #191414;
}