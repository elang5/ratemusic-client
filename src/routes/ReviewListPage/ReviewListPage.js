import React, { useEffect } from "react";
import { connect } from "react-redux";
import { setReviews, setAlbumName, setAlbumURL, setError } from "../../actions";
import { Link } from "react-router-dom";
import AlbumApiService from "../../services/albums-api-service";
import ReviewItem from "../../components/ReviewItem/ReviewItem";
import "./ReviewListPage.css";

function ReviewListPage(props) {
  useEffect(() => {
    const { albumId } = props.match.params;
    AlbumApiService.getAlbumReviews(albumId)
      .then(res => props.dispatch(setReviews([...res])))
      .catch(err => props.dispatch(setError(err.error)));
    AlbumApiService.getAlbum(albumId)
      .then(album => {
        props.dispatch(setAlbumName(album.name));
        props.dispatch(setAlbumURL(album.external_urls.spotify));
      })
      .catch(err => props.dispatch(setError(err.error)));
  }, []);

  const { reviews, error, album_name, album_url } = props;
  const { albumId } = props.match.params;
  return (
    <section className="album-reviews-page">
      <div className="buttons">
        <Link className="post-review" to={`/${albumId}/reviews`}>
          <button className="review-btn">Post a Review!</button>
        </Link>
        <a
          className="album-listen"
          href={album_url}
          target="_blank"
          rel="noopener noreferrer"
        >
          <button className="listen-btn">{`Listen to ${album_name}`}</button>
        </a>
      </div>
      {error && <p className="no-reviews-error">{error}</p>}
      <ul className="review-list">
        {reviews &&
          reviews.map(review => <ReviewItem key={review.id} review={review} />)}
      </ul>
    </section>
  );
}

const mapStateToProps = state => ({
  reviews: state.reviews,
  album_name: state.album_name,
  album_url: state.album_url
});

export default connect(mapStateToProps)(ReviewListPage);

ReviewListPage.defaultProps = {
  match: { params: {} }
};
