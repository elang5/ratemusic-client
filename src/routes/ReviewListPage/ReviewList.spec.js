import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReviewListPage from './ReviewListPage'

describe(`ReviewListPage Component`, () => {
  it(`renders the ReviewListPage component`, () => {
    const wrapper = shallow(<ReviewListPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})