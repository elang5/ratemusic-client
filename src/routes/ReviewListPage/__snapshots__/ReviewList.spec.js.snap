// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewListPage Component renders the ReviewListPage component 1`] = `
<section
  className="album-reviews-page"
>
  <div
    className="buttons"
  >
    <Link
      className="post-review"
      to="/undefined/reviews"
    >
      <button
        className="review-btn"
      >
        Post a Review!
      </button>
    </Link>
    <a
      className="album-listen"
      href=""
    >
      <button
        className="listen-btn"
      >
        Listen to 
      </button>
    </a>
  </div>
  <ul
    className="review-list"
  />
</section>
`;
